### DISCUSSION TITLE
In the SharedMemory example, sometimes the publisher and subscriber topic cannot be matched

### DISCUSSION THREAD

**Discussion by foreverkid** [eProsima/Fast-DDS#3727] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

When a publisher and subscriber have the same topic, they should match every time.

### Current behavior

Sometimes publisher and subscriber cannot match and communicate

### Steps to reproduce

- Compile the QNX version of example fast-dds/examples/C++/DDS/HelloWorldExampleSharedMem
- Open two terminals in the qnx system to run ./DDSHelloWorldExampleSharedMem publisher  and ./DDSHelloWorldExampleSharedMem subscriber respectively
- Keep restarting the QNX system and run the previous step
- Sometimes publisher and subscriber cannot matched and communicate

### Fast DDS version/commit

2.6.X

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

UDPv4, Shared Memory Transport (SHM), Zero copy

### Additional context

Platform: QNX7

I turned on the debug function of dds and compared the difference between connected and unconnected. Before printing out the "Subscriber matched." message, they differ in the following

Connected:
01.0f.ce.6a.1e.90.de.ae.01.00.00.00|0.0.1.c1 at MTTLoc: {**UNICAST:[SHM:[]:7414]} DefLoc:{UNICAST:[SHM:[]:7415]**} -> Function onNewCacheChangeAdded
**[RTPS_TRANSPORT_SHM Info] (ID:5) Opening fastrtps_port7414 -> Function open_port_internal
[RTPS_TRANSPORT_SHM Info] (ID:5) Port 7414 (0d449125890005a2) OpenedWrite -> Function open_port_internal**

Disconnected:
01.0f.ea.21.1e.10.f4.ef.01.00.00.00|0.0.1.c1 at MTTLoc: {} DefLoc:{} -> Function onNewCacheChangeAdded

### XML configuration file

```xml
No response
```

### Relevant log output

```shell
No response
```

### Network traffic capture

No response

**Comment by JLBuenoLopez** [comment]
Thanks for your contribution @foreverkid,

[QNX is not a Tier 1 platform](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md). Currently is considered Tier 3 supported and consequently, as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support), I am moving this issue to the Unofficial support discussion forum.

Recently, some long standing Shared Memory issues where fixed related to ports stuck due to non-cleanly closed applications. These fixes have been backported to 2.6.x but have not been merged yet (#3695, #3699, #3711). You might try with a branch taking those fixes and see if your issue is still happening.
